java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/utils/Json.java

```diff
@@ -18,89 +18,207 @@
 
   private final JSONObject document;
 
+  /**
+   * Creates a new {@code Json} instance.
+   */
   public Json() {
     document = new JSONObject();
   }
 
+  /**
+   * Convert the {@link org.json.JSONObject} to a {@link java.util.Set} object.
+   *
+   * @return The json document as a {@link java.util.Set} object.
+   */
   public Set<String> keySet() {
     return document.keySet();
   }
 
+  /**
+   * Creates a new {@code Json} instance from the provided {@code JSONObject}.
+   *
+   * @param jsonObject The {@link org.json.JSONObject} object to convert to {@code Json}.
+   */
   public Json(final JSONObject jsonObject) {
     if (jsonObject == null) {
       throw new IllegalArgumentException("jsonObject is null!");
     }
     document = jsonObject;
   }
 
+  /**
+   * Creates a new {@code Json} instance from the provided byte array.
+   *
+   * @param utf8Json An array of bytes to be converted to {@code Json}.
+   */
   public Json(final byte[] utf8Json) {
     document = convertUtf8ToJson(utf8Json);
   }
 
+  /**
+   * Gets the {@code Json} associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key.
+   */
   public Json getJson(final String key) {
     return new Json(document.getJSONObject(key));
   }
 
+  /**
+   * Gets the @code Json} associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return An {@link Optional} Json value if the key exists, else null
+   */
   public Optional<Json> getOptionalJson(final String key) {
     return Optional.ofNullable(document.optJSONObject(key)).map(Json::new);
   }
 
+  /**
+   * Gets the string value associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key, as a string.
+   */
   public String getString(final String key) {
     return document.getString(key);
   }
 
+  /**
+   * Converts the key into a newly-allocated byte array.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key, as a bytes array.
+   */
   public byte[] getBytes(final String key) {
     return Base64.getDecoder().decode(document.getString(key));
   }
 
+  /**
+   * Retrieves the long value associated with a given key and converts it an instance of {@link Instant} using seconds
+   * from the epoch of 1970-01-01T00:00:00Z.
```

Comment:
```suggestion
   * Retrieves the long value associated with a given key and converts it an instance of {@link Instant} using unix epoch.
```

Reply:
Used the same text as the JavaDoc of `ofEpochSecond`. Think this should be fine.